### PR TITLE
Armor of the Imperiya : Giving some real life to the NRI surplus armor

### PR DESCRIPTION
**Pull request by Paxilmaniac**
## About The Pull Request

image

image
image

(The colors and looks in the above images are not final, but take a peek anyways :3)

Today on how I learned to stop hating and start going "this could be cooler". (Currently a work in progress) Resprite and rework of the NRI surplus armor you can get from cargo.

So, what's new?

Aside from the updated looks, there's a bit more of a 'surplus' flair to the armor now. The current plans are that nri armor gotten through the imperial surplus crate will come as random variations of coloring and paint scheme, to give that feel of literally ordering whatever surplus stuff the NRI had on offer. Aside from just different colors, there are also smaller variations! Some helmets might come missing their glass visor, or their chin strap... And the vests might come missing some of their extra plates!

## How This Contributes To The Skyrat Roleplay Experience

It was a really funny idea that turned into something kinda cool actually. I think we could be doing something really funny and unique with ordering surplus NRI stuff, and overall I think the new sprites are more befitting of surplus armor that came from a border war happening in the interstellar age, don't you think?

## Proof of Testing

If its not shown above, its shown here:

image
(the backpack was fixed after this image was taken)

image
image
image

## Changelog

:cl:
imageadd: The NRI surplus armor is getting a pretty big resprite and reflavoring, ordering surplus armor in the surplus crates might give you a bunch of mismatched colors you know!
add: Izlishek Company Military Supplier has stopped selling the old NRI armor vests, in exchange for selling tactical floor tile grey variations of the new CIN armor and uniform set
balance: This does, technically, mean a balance change. For the bit of lore behind the armor, and the actual armor values to line up, and for the vests to not just be "sec vests but better in every way", they've lost a majority of their (barely existent) energy and laser protection, in exchange for a little extra bomb and wound armor.
/:cl:

### CODE REVIEW

**Comment by Stalkeros2** [comment]
Wooo I won't have to commission a spriter now thank you so much unironically.

File: modular_skyrat/modules/novaya_ert/code/surplus_armor.dm [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	name = "surplus nri vest"
+	desc = "Wait a minute, there should be something here!."
+	icon = 'modular_skyrat/modules/novaya_ert/icons/surplus_armor/surplus_armor.dmi'
+	worn_icon = 'modular_skyrat/modules/novaya_ert/icons/surplus_armor/surplus_armor.dmi'
+	icon_state = "vest_basic"
+	armor_type = /datum/armor/russian_nri
+	supports_variations_flags = CLOTHING_NO_VARIATION
+
+	/// Weighted list for determining if the vest will have its extra plates or not
+	var/static/list/extra_plates_yeah_or_nah = list(
```
Comment (Paxilmaniac) [inline modular_skyrat/modules/novaya_ert/code/surplus_armor.dm]: shoutout to the variable of all time

**Comment by HoodMilk** [comment]
Power Ranger Sec

It looks real nice though

File: modular_skyrat/modules/cargo/code/packs.dm [hunk 2]
```diff
@@ -695,34 +695,36 @@
...
+		/obj/item/storage/backpack/industrial/cin_surplus/random_color = 3,
+		/obj/item/storage/belt/military/cin_surplus/random_color = 3,
+		/obj/item/clothing/gloves/tackler/combat = 3,
+		/obj/item/clothing/under/syndicate/rus_army/cin_surplus/random_color = 3,
+		/obj/item/clothing/shoes/jackboots = 3,
+		/obj/item/clothing/mask/gas/hecu2 = 3,
+		/obj/item/clothing/mask/balaclavaadjust = 3,
+	)
+
+/datum/supply_pack/security/armory/russian/fill(obj/structure/closet/crate/C)
```
Comment (Zonespace27) [inline modular_skyrat/modules/cargo/code/packs.dm]: one letter var

File: modular_skyrat/modules/novaya_ert/code/surplus_armor.dm [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+	greyscale_colors = CIN_WINTER_COLORS
+	armor_type = /datum/armor/cin_surplus_armor
+	supports_variations_flags = CLOTHING_SNOUTED_VARIATION_NO_NEW_ICON
+
+	/// Controls what helmet accessories will be present in a weighted format
+	var/static/list/accessories_weighted_list = list(
+		"plain" = 15,
+		"strap" = 10,
+		"glass" = 10,
+		"both" = 5,
```
Comment (Zonespace27) [inline modular_skyrat/modules/novaya_ert/code/surplus_armor.dm]: defines

File: modular_skyrat/modules/novaya_ert/code/surplus_armor.dm [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+	)
+
+/obj/item/clothing/suit/armor/vest/cin_surplus_vest/Initialize(mapload)
+	. = ..()
+
+	generate_random_accessories()
+
+/// Decides if the armor vest should have its extra plates or not
+/obj/item/clothing/suit/armor/vest/cin_surplus_vest/proc/generate_random_accessories()
+	if(pick_weight(extra_plates_yeah_or_nah) == "has_da_plates")
```
Comment (Zonespace27) [inline modular_skyrat/modules/novaya_ert/code/surplus_armor.dm]: can't this just be a prob(33) instead of needing the static list and all that

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/cargo/code/packs.dm (hunk 2 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/cargo/code/packs.dm]: i love copy and pasting cargo code :)

File: modular_skyrat/modules/novaya_ert/code/surplus_armor.dm (hunk 1 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/novaya_ert/code/surplus_armor.dm]: there is no more variable of all time

**Review by Ebin-Halcyon — APPROVED** [review]
Looks fine.

**Review by Zonespace27 — APPROVED** [review]
lgtm

**Comment by masline147** [comment]
Now it is like bulletpruf armor, but it is worse. Whoooooooo, here's the update. Why didn't you say that this is how the protection of this armor will fall? She's useless now

**Pull request merged**